### ISSUE TITLE
Incorrect port being found by Restler.php

### ISSUE THREAD

**Issue by Tommyf** [Luracast/Restler#479]
I was just running through the examples and noticed some did not work out of the box. Specifically, any that use the API explorer would fail as a request to **http://host:port/restler/public/examples/_008_documentation/resources.json** would return the incorrect basePath in the response.

This is occuring because my Apache install is listening on a non-standard port, but the 'ServerName' does not necessarily include the port. Consider this:

**Apache Config**

    <VirtualHost *:4242>
        ServerName restlertest

If you then visit:
**http://restlertest:4242/restler/public/examples/_008_documentation/resources.json**

the resultant **basePath** would be:

**http://restlertest/restler/public/examples/_008_documentation/resources.json,**

This is fixable by changing the **ServerName** to **restlertest:4242**, but then if you visit through a reverse proxy that is listening on port 80, the **basePath** would still return the URL with port 4242.

It is a fairly common scenario to have an internet facing reverse proxy listen on port 80/443 and for the back-end server to host the site on a different port, especially in the Enterprise. In these instances, leaving ServerName set to restlertest would work for the reverse-proxied clients, but not for internal users testing directly.

**Enough of the background, on to the fix.**

Unfortunately, PHP's **$_SERVER['SERVER_PORT']** is an unreliable way to retrieve the port that the client is using to connect with. Another way to find the port is to see if it was included as a part of the HOST header. This can be done in PHP using **$_SERVER['HTTP_HOST']**

I propose fixing using the following method:

Retrieve the host header. If it contains a '**:**' then the port has been explicitly specified by the client, so grab it from the host string. Otherwise use with the existing method of port retrieval.

I have fixed and tested my local copy but I have not committed any fixes on github before. Can I just go ahead and do it?

Thanks,

Tom...

**Comment by Tommyf** [comment]
More info - I was testing this at home on Apache 2.4 and I am not seeing the same behaviour I was seeing at work on Apache 2.2. I tried setting UseCanonicalName to 'On' and I saw the same (bad) behaviour. I double checked my Apache 2.2 and it behaves the same with UseCanonicalName set to On or Off.

It looks to me like a bug in Apache 2.2.3 (RHEL 5.11). It it worth putting the fix for this in the Restler code? I doubt many people will be installing this on such an antiquated Apache/OS.

But then again, the fix will not break newer installations either.
